--- cron-apt.orig	2007-07-28 09:08:16.000000000 +0200
+++ cron-apt	2007-07-28 09:26:10.000000000 +0200
@@ -142,7 +142,11 @@
 CONFIGDIRNAME="$(echo $CONFIG | sed 's|/|_-_|g')"
 TMPDIR=$(mktemp -d -t cron-apt.XXXXXX)
 if [ $? -ne 0 ]; then
-    echo "Error: Can not create a safe temporary directory."
+    echo >&2 "Error: Can not create a safe temporary directory."
+    exit 1
+fi
+if [ "$(( $(stat --file-system --format=%S $TMPDIR) * $(stat --file-system --format=%a $TMPDIR) / 1024 ))" -lt "10000" ]; then
+    echo >&2 "Error: Not enough free space in $TMPDIR."
     exit 1
 fi
 
