--- cron-apt.orig	2011-02-09 15:17:27.000000000 +0100
+++ cron-apt.new	2011-02-09 15:17:02.000000000 +0100
@@ -350,7 +350,7 @@
 	    sed -e "s/#.*$//;" | \
 	    grep -v "^[[:space:]]*$" | {
 		while read LINE ; do
-		    echo "CRON-APT LINE: $APTCOMMAND $LINE" > "$TEMP"
+		    echo "CRON-APT LINE: $APTCOMMAND $OPTIONS $LINE" > "$TEMP"
 		    UMASK_SAVE=$(umask)
 		    umask $UMASK_APT
 		    $APTCOMMAND $OPTIONS $LINE >> $TEMP 2>&1
