diff -Nru cron-apt-0.4.11.orig/conf/config cron-apt-0.4.11/conf/config
--- cron-apt-0.4.11.orig/conf/config	2006-12-14 14:59:01.000000000 +0100
+++ cron-apt-0.4.11/conf/config	2006-12-14 15:13:30.000000000 +0100
@@ -139,6 +139,11 @@
 # aptitude 0.3.4.
 #  OPTIONS="-o quiet=1 -o APT::Get::AllowUnauthenticated=true -o aptitude::Cmdline::ignore-trust-violations=yes"
 
+# additional APT configuration file that is loaded first. This can be set in
+# order to use a completely different APT configuration for cron-apt. See the
+# /usr/share/doc/cron-apt/README and apt.conf(5) for details
+# export APT_CONFIG=/etc/apt/cron.apt.paths
+
 # Do not run the command, if there is an error in the previous run (default).
 # Value: error   (do not run if there is an error on last run)
 #                (else always run, remove previous error file and run)
diff -Nru cron-apt-0.4.11.orig/README cron-apt-0.4.11/README
--- cron-apt-0.4.11.orig/README	2006-12-14 14:59:01.000000000 +0100
+++ cron-apt-0.4.11/README	2006-12-14 15:10:23.000000000 +0100
@@ -40,6 +40,26 @@
 
  OPTIONS="-q -o Dir::Etc::SourceList=/etc/apt/security.sources.list"
 
+Alternate APT main configuration
+--------------------------------
+
+There are cases when you need the APT configuration for use with cron-apt to
+differ substantially from your standard configuration. For example, cron-apt
+will not work with the default configuration of APT, when apt-listbugs is
+installed, because this adds a hook that expects keyboard interactivity.
+
+Unfortunately, the apt-tools do not accept Dir::Etc::Main and Dir::Etc::Parts
+settings on the command-line. In order to use different paths for them, you
+have to create another APT configuration file containing eg:
+
+ Dir::Etc::Main "cron.apt.conf";
+ Dir::Etc::Parts "cron.apt.conf.d";
+
+Then uncomment the following line in your cron-apt configuration file and
+point it to your newly created APT configuration file:
+
+ export APT_CONFIG=/etc/apt/cron.apt.paths
+
 Development and documentation
 -----------------------------
 
