--- cron-apt	2003-07-22 12:57:05.000000000 +0200
+++ cron-apt-ri	2003-07-22 13:00:00.000000000 +0200
@@ -25,9 +25,13 @@
 ############################# arguments #######################################
 
 if [ "$1" = "--help" ] ; then
-    echo "USAGE: cron-apt [configfile]"
+    echo "USAGE: cron-apt [-i] [configfile]"
     exit 0
 fi
+if [ "$1" = "-i" ] ; then
+    RUNIMMEDIATELY="yes"
+    shift
+fi
 CONFIG=$1
 if [ -z "$CONFIG" ] ; then
     CONFIG=/etc/cron-apt/config
@@ -66,6 +70,10 @@
     echo "The config file $CONFIG does not exist, using defaults."
 fi
 
+if test "$RUNIMMEDIATELY" = "yes"; then
+	RUNSLEEP=0
+fi
+
 ############################## functions ######################################
 
 # Send the mail to the system administrator. Will only be sent if there is
@@ -112,7 +120,7 @@
 
 ############################### sleep #########################################
 
-if [ -n "$RUNSLEEP" ] ; then
+if [ "$RUNSLEEP" -gt 0 ] ; then
     sleep $(($RANDOM % $RUNSLEEP))
 fi
 
