From: Marc Haber <mh+debian-bugs@zugschlus.de>

diff -urN cron-apt/conf/config cron-apt-0.0.34.0/conf/config
--- cron-apt/conf/config	2004-08-13 08:24:47.000000000 +0200
+++ cron-apt-0.0.34.0/conf/config	2004-09-19 22:27:24.000000000 +0200
@@ -48,6 +48,10 @@
 # the $SYSLOGON directive.
 # SYSLOGONMSGSDIR=/etc/cron-apt/syslogonmsgs
 
+# Value: ""       (warn if dotlockfile not installed)
+#        "nowarn" (don't give warning if dotlockfile not installed)
+# NOLOCKWARN=""
+
 # The file that contains error messages.
 # ERROR=/var/log/cron-apt/error
 
diff -urN cron-apt/debian/changelog cron-apt-0.0.34.0/debian/changelog
--- cron-apt/debian/changelog	2004-09-15 13:34:43.000000000 +0200
+++ cron-apt-0.0.34.0/debian/changelog	2004-09-19 22:36:19.000000000 +0200
@@ -1,3 +1,9 @@
+cron-apt (0.0.34.0) unstable; urgency=low
+
+  * add warning if dotlockfile not found
+
+ -- Marc Haber <mh+debian-packages@zugschlus.de>  Sun, 19 Sep 2004 22:36:05 +0200
+
 cron-apt (0.0.34) unstable; urgency=low
 
   * Made locking optional but recommended. Also added recommends on
diff -urN cron-apt/src/cron-apt cron-apt-0.0.34.0/src/cron-apt
--- cron-apt/src/cron-apt	2004-09-15 13:34:43.000000000 +0200
+++ cron-apt-0.0.34.0/src/cron-apt	2004-09-19 22:32:00.000000000 +0200
@@ -135,6 +135,7 @@
 MAILMSGDIR="/etc/cron-apt/mailmsg.d"
 MAILONMSGSDIR="/etc/cron-apt/mailonmsgs"
 SYSLOGONMSGSDIR="/etc/cron-apt/syslogonmsgs"
+NOLOCKWARN=""
 ERRORMSGDIR="/etc/cron-apt/errormsg.d"
 SYSLOGMSGDIR="/etc/cron-apt/syslogmsg.d"
 LOGMSGDIR="/etc/cron-apt/logmsg.d"
@@ -381,6 +382,10 @@
 			fi
 			dotlockfile -u $LOCKFILE
 		    else
+		    	if [ -z "$NOLOCKWARN" ]; then
+			    echo >  $TEMP "WARNING: dotlockfile not installed. If you don't want to see this"
+			    echo >> $TEMP "         Warning any more, set NOLOCKWARN in the config file."
+			fi
 			apt-get $OPTIONS $LINE >> $TEMP 2>&1
 			RET=$?
 		    fi
