diff -ur cron-apt-0.4.7.orig/conf/config cron-apt-0.4.7/conf/config
--- cron-apt-0.4.7.orig/conf/config	2006-03-05 19:09:17.000000000 +0000
+++ cron-apt-0.4.7/conf/config	2006-07-10 16:59:57.000000000 +0100
@@ -122,7 +122,7 @@
 #        append  (append to the output)
 #	 only	 (only show the diff, not the output itself)
 #                (else do nothing)
-DIFFONCHANGES=prepend
+# DIFFONCHANGES=prepend
 
 # General apt options that will be passed to all APTCOMMAND calls.
 # Use "-o quiet" instead of "-q" for aptitude compatibility.
diff -ur cron-apt-0.4.7.orig/src/cron-apt cron-apt-0.4.7/src/cron-apt
--- cron-apt-0.4.7.orig/src/cron-apt	2006-04-30 17:21:27.000000000 +0100
+++ cron-apt-0.4.7/src/cron-apt	2006-07-10 16:59:21.000000000 +0100
@@ -197,6 +197,12 @@
 # Ignore lines matching this regexp to determine whether changes occurred
 # for MAILON="changes".
 DIFFIGNORE=""
+# What to do with the diff when *ON=changes.
+# Value: prepend (prepend to the output)
+#        append  (append to the output)
+#        only    (only show the diff, not the output itself)
+#                (else do nothing)
+DIFFONCHANGES=prepend
 
 export DEBIAN_FRONTEND="noninteractive"
 export LANG="C"
