From: Marc Haber <mh+debian-bugs@zugschlus.de>

--- cron-apt~	2004-08-20 21:41:30.000000000 +0000
+++ cron-apt	2004-09-05 15:57:20.000000000 +0000
@@ -272,6 +272,26 @@
     createdivinfo "$ERROR" "$TEMP" "$ERRORMSGDIR/$1" "$RUNERROR" "$ACTIONERROR"
 }
 
+# run-parts emulation, stolen from Branden's /etc/X11/Xsession
+run_parts () {
+	# reset LC_COLLATE
+	unset LANG LC_COLLATE LC_ALL
+
+	if [ -z "$1" ]; then
+		echo >&2 "$0: internal run_parts called without an argument"
+	fi
+	if [ ! -d "$1" ]; then
+		echo >&2 "$0: internal run_parts called, but $1 does not exist or is not a directory."
+	fi
+	for F in $(ls $1); do
+		if expr "$F" : '[[:alnum:]_-]\+$' > /dev/null 2>&1; then
+			if [ -f "$1/$F" ] ; then
+				echo "$1/$F"
+			fi
+		fi
+	done;
+}
+
 ############################### check #########################################
 
 if [ -e "$ERROR" ] && [ "$DONTRUN" = "error" ] ; then
@@ -309,17 +329,8 @@
 ############################### script ########################################
 # Always run onexit before exit.
 
-for ACTIONF in $(ls "$ACTIONDIR") ; do
-    if echo "$ACTIONF" | grep "~" > /dev/null 2>&1 ; then
-	# Temporary file, skipping.
-	:
-    elif echo "$ACTIONF" | grep ".dpkg-dist" > /dev/null 2>&1 ; then
-	# dpkg-dist file
-	:
-    elif echo "$ACTIONF" | grep ".dpkg-old" > /dev/null 2>&1 ; then
-	# dpkg-old file
-	:
-    elif [ -f "$ACTIONDIR/$ACTIONF" ] ; then
+for ACTION in $(run_parts "$ACTIONDIR") ; do
+	ACTIONF=$(echo $ACTION | sed "s|$ACTIONDIR||")
 	if [ -f "$ACTIONCONFDIR/$ACTIONF" ] ; then
 	    . "$ACTIONCONFDIR/$ACTIONF"
 	fi
@@ -452,7 +463,6 @@
 	    fi
 	    rm -f "$STATUS"
 	fi
-    fi
 done
 
 onexit
